From b7e63ed39e3383e78f146212d9c8b4eb0eea1a7d Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Mon, 12 Mar 2018 20:05:02 +0900
Subject: [PATCH 117/909] clk: renesas: cpg-mssr: Add DEF_GEN3_Z macro

commit abb24ccb7cfaaa0e3ca80ddd715e77a1c6f6e6d0 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

The division value of z clock divider differs for each SoC. This patch
add a macro that specifies the division value of z clock divider in
struct cpg_core_clk[] for each SoC.

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/renesas/rcar-gen3-cpg.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/clk/renesas/rcar-gen3-cpg.h b/drivers/clk/renesas/rcar-gen3-cpg.h
index 57768a3b0523..8bca329888d6 100644
--- a/drivers/clk/renesas/rcar-gen3-cpg.h
+++ b/drivers/clk/renesas/rcar-gen3-cpg.h
@@ -2,6 +2,7 @@
  * R-Car Gen3 Clock Pulse Generator
  *
  * Copyright (C) 2015-2018 Glider bvba
+ * Copyright (C) 2018 Renesas Electronics Corp.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -48,6 +49,9 @@ enum rcar_gen3_clk_types {
 	DEF_BASE(_name, _id, CLK_TYPE_GEN3_RCKSEL,	\
 		 (_parent0) << 16 | (_parent1),	.div = (_div0) << 16 | (_div1))
 
+#define DEF_GEN3_Z(_name, _id, _type, _parent, _div)	\
+	DEF_BASE(_name, _id, _type, _parent, .div = _div)
+
 struct rcar_gen3_cpg_pll_config {
 	u8 extal_div;
 	u8 pll1_mult;
-- 
2.17.1

